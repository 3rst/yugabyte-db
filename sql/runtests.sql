\unset ECHO
\i test_setup.sql
SET client_min_messages = warning;

-- $Id$

CREATE SCHEMA whatever;
CREATE TABLE whatever.foo ( id serial primary key );

-- Make sure we get test function names.
SET client_min_messages = notice;

CREATE OR REPLACE FUNCTION whatever.startup() RETURNS SETOF TEXT AS $$
    SELECT pass('starting up');
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.startup_more() RETURNS SETOF TEXT AS $$
    SELECT pass('starting up some more');
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.setup() RETURNS SETOF TEXT AS $$
    SELECT pass('setup')
    UNION
    SELECT is( MAX(id), NULL, 'Should be nothing in the test table') FROM whatever.foo;
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.setup_more() RETURNS SETOF TEXT AS $$
    SELECT pass('setup more');
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.teardown() RETURNS SETOF TEXT AS $$
    SELECT pass('teardown');
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.teardown_more() RETURNS SETOF TEXT AS $$
    SELECT pass('teardown more');
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.shutdown() RETURNS SETOF TEXT AS $$
    SELECT pass('shutting down');
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.shutdown_more() RETURNS SETOF TEXT AS $$
    SELECT pass('shutting down more');
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.test_this() RETURNS SETOF TEXT AS $$
    SELECT pass('simple pass') AS foo
    UNION SELECT pass('another simple pass')
    ORDER BY foo ASC;
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever.testplpgsql() RETURNS SETOF TEXT AS $$
BEGIN
    RETURN NEXT pass( 'plpgsql simple' );
    RETURN NEXT pass( 'plpgsql simple 2' );
    INSERT INTO whatever.foo VALUES(1);
    RETURN NEXT is( MAX(id), 1, 'Should be a 1 in the test table') FROM whatever.foo;
END;
$$ LANGUAGE plpgsql;

CREATE OR REPLACE FUNCTION whatever.testz() RETURNS SETOF TEXT AS $$
    SELECT is( MAX(id), NULL, 'Late test should find nothing in the test table') FROM whatever.foo;
$$ LANGUAGE SQL;

CREATE OR REPLACE FUNCTION whatever."test ident"() RETURNS SETOF TEXT AS $$
BEGIN
    RETURN NEXT pass( 'ident' );
    RETURN NEXT pass( 'ident 2' );
END;
$$ LANGUAGE plpgsql;

-- Run the actual tests. Yes, it's a one-liner!
SELECT * FROM runtests('whatever'::name);

ROLLBACK;
