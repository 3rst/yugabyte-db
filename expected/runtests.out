\unset ECHO
ok 1 - starting up
ok 2 - starting up some more
# whatever."test ident"()
ok 3 - setup
ok 4 - Should be nothing in the test table
ok 5 - setup more
ok 6 - ident
ok 7 - ident 2
ok 8 - teardown
ok 9 - teardown more
# whatever.test_this()
ok 10 - setup
ok 11 - Should be nothing in the test table
ok 12 - setup more
ok 13 - simple pass
ok 14 - another simple pass
ok 15 - teardown
ok 16 - teardown more
# whatever.testplpgsql()
ok 17 - setup
ok 18 - Should be nothing in the test table
ok 19 - setup more
ok 20 - plpgsql simple
ok 21 - plpgsql simple 2
ok 22 - Should be a 1 in the test table
ok 23 - teardown
ok 24 - teardown more
# whatever.testz()
ok 25 - setup
ok 26 - Should be nothing in the test table
ok 27 - setup more
ok 28 - Late test should find nothing in the test table
ok 29 - teardown
ok 30 - teardown more
ok 31 - shutting down
ok 32 - shutting down more
1..32
